### PR TITLE
GRIDEDIT-1448: Add CurvilinearDeleteExterior/Interior

### PR DESCRIPTION
**Pull request by lucacarniato**

### CODE REVIEW

File: src/MeshKernelNET/Api/IMeshKernelApi.cs [hunk 1]
```diff
@@ -186,6 +186,37 @@ int CurvilinearComputeTransfiniteFromTriangle(int meshKernelId,
...
+        /// <param name="yUpperRightCorner">The y coordinate of the upper right corner of the block to delete</param>
+        /// <returns>Error code</returns>
+        int CurvilinearDeleteExterior(int meshKernelId,
+            double xLowerLeftCorner,
+            double yLowerLeftCorner,
+            double xUpperRightCorner,
+            double yUpperRightCorner);
+
+        /// <summary>
+        /// Delete the exterior part of a curvilinear grid
```
Comment (andreasbuykx) [inline src/MeshKernelNET/Api/IMeshKernelApi.cs:206]: suggestion to rephrase: ... exterior part of a rectangle in the curvilinear grid

File: src/MeshKernelNET/Api/IMeshKernelApi.cs [hunk 2]
```diff
@@ -186,6 +186,37 @@ int CurvilinearComputeTransfiniteFromTriangle(int meshKernelId,
         /// <returns>Error code</returns>
         int CurvilinearConvertToMesh2D(int meshKernelId);
 
+
+        /// <summary>
+        /// Delete the interior part of a curvilinear grid
```
Comment (andreasbuykx) [inline src/MeshKernelNET/Api/IMeshKernelApi.cs:191]: suggestion to rephrase: ... interior part of a rectangle in the curvilinear grid

File: src/MeshKernelNET/Native/MeshKernelDll.cs [hunk 3]
```diff
@@ -219,6 +219,38 @@ internal static extern int CurvilinearComputeTransfiniteFromTriangle([In] int me
         [DllImport(MeshKernelDllName, EntryPoint = "mkernel_curvilinear_convert_to_mesh2d", CallingConvention = CallingConvention.Cdecl)]
         public static extern int CurvilinearConvertToMesh2D([In] int meshKernelId);
 
+        /// <summary>
+        /// Delete the interior part of a curvilinear grid
```
Comment (andreasbuykx) [inline src/MeshKernelNET/Native/MeshKernelDll.cs:223]: rephrase suggested like above

File: src/MeshKernelNET/Native/MeshKernelDll.cs [hunk 4]
```diff
@@ -219,6 +219,38 @@ internal static extern int CurvilinearComputeTransfiniteFromTriangle([In] int me
...
+        /// <returns>Error code</returns>
+        [DllImport(MeshKernelDllName, EntryPoint = "mkernel_curvilinear_delete_interior", CallingConvention = CallingConvention.Cdecl)]
+        public static extern int CurvilinearDeleteInterior([In] int meshKernelId,
+                                                           [In] double xLowerLeftCorner,
+                                                           [In] double yLowerLeftCorner,
+                                                           [In] double xUpperRightCorner,
+                                                           [In] double yUpperRightCorner);
+
+        /// <summary>
+        /// Delete the exterior part of a curvilinear grid
```
Comment (andreasbuykx) [inline src/MeshKernelNET/Native/MeshKernelDll.cs:239]: rephrase suggested like above

**Review by andreasbuykx — APPROVED** [review]
Consider changing or extending the interface such that it is possible to pass in index pairs defining the vertices. That way a spatial search can be avoided.

**Comment by lucacarniato** [comment]
> Consider changing or extending the interface such that it is possible to pass in index pairs defining the vertices. That way a spatial search can be avoided.

i prefer to create a separate item for this. for now I substituted the 4 coordinates with a bounding box

**Pull request merged**